modules/gdscript/language_server/gdscript_extend_parser.cpp

```diff
@@ -0,0 +1,237 @@
+		symbol.detail = m.data_type.to_string();
+		symbol.deprecated = false;
+		const int line = m.line - 1;
+		symbol.range.start.line = line;
+		symbol.range.start.character = lines[line].length() - lines[line].strip_edges(true, false).length();
+		symbol.range.end.line = line;
+		symbol.range.end.character = lines[line].length();
+		symbol.selectionRange.start.line = symbol.range.start.line;
+
+		r_symbol.children.push_back(symbol);
+	}
+
+	for (int i = 0; i < p_class->_signals.size(); ++i) {
+		const GDScriptParser::ClassNode::Signal &signal = p_class->_signals[i];
+
+		lsp::DocumentSymbol symbol;
+		symbol.name = signal.name;
+		symbol.kind = lsp::SymbolKind::Event;
+		symbol.deprecated = false;
+		const int line = signal.line - 1;
+		symbol.range.start.line = line;
+		symbol.range.start.character = lines[line].length() - lines[line].strip_edges(true, false).length();
+		symbol.range.end.line = symbol.range.start.line;
+		symbol.range.end.character = lines[line].length();
+		symbol.selectionRange.start.line = symbol.range.start.line;
+
+		r_symbol.children.push_back(symbol);
+	}
+
+	for (Map<StringName, GDScriptParser::ClassNode::Constant>::Element *E = p_class->constant_expressions.front(); E; E = E->next()) {
+		lsp::DocumentSymbol symbol;
+		symbol.name = E->key();
+		symbol.kind = lsp::SymbolKind::Constant;
+		symbol.deprecated = false;
+		const int line = E->get().expression->line - 1;
+		symbol.range.start.line = line;
+		symbol.range.start.character = E->get().expression->column;
+		symbol.range.end.line = symbol.range.start.line;
+		symbol.range.end.character = lines[line].length();
+		symbol.selectionRange.start.line = symbol.range.start.line;
+
+		r_symbol.children.push_back(symbol);
+	}
+
+	for (int i = 0; i < p_class->functions.size(); ++i) {
+		const GDScriptParser::FunctionNode *func = p_class->functions[i];
+		lsp::DocumentSymbol symbol;
+		parse_function_symbol(func, symbol);
+		r_symbol.children.push_back(symbol);
+	}
+
+	for (int i = 0; i < p_class->static_functions.size(); ++i) {
+		const GDScriptParser::FunctionNode *func = p_class->static_functions[i];
+		lsp::DocumentSymbol symbol;
+		parse_function_symbol(func, symbol);
+		r_symbol.children.push_back(symbol);
+	}
+
+	for (int i = 0; i < p_class->subclasses.size(); ++i) {
+		const GDScriptParser::ClassNode *subclass = p_class->subclasses[i];
+		lsp::DocumentSymbol symbol;
+		parse_class_symbol(subclass, symbol);
+		r_symbol.children.push_back(symbol);
+	}
+}
+
+void ExtendGDScriptParser::parse_function_symbol(const GDScriptParser::FunctionNode *p_func, lsp::DocumentSymbol &r_symbol) {
+	r_symbol.name = p_func->name;
+	r_symbol.kind = lsp::SymbolKind::Function;
+	r_symbol.detail = p_func->get_datatype().to_string();
+	r_symbol.deprecated = false;
+	const int line = p_func->line - 1;
+	r_symbol.range.start.line = line;
+	r_symbol.range.start.character = p_func->column;
+	r_symbol.range.end.line = MAX(p_func->body->end_line - 2, p_func->body->line);
```

Comment:
As the parser always give me a larger result.

Reply:
Is this because of the space between the functions (as in: the end line is the beginning of the next function) or the parser just gives an off-by-two value? Either way, I would prefer to fix this in the GDScript parser if possible.